src/message_queue.rs

```diff
@@ -6,85 +6,54 @@
-    identifier: Identifier,
-}
-
-#[derive(Serialize, Deserialize, Clone)]
-pub(crate) struct MessageQueue(HashMap<Vec<u8>, Vec<Message>>);
+use std::collections::HashMap;
 
-impl Default for MessageQueue {
-    fn default() -> Self {
-        Self::new()
-    }
-}
+/// A type for storing a queue of [`Message`]s by [`MessageType`].
+#[derive(Clone, Default)]
+pub(crate) struct MessageQueue(HashMap<MessageType, Vec<Message>>);
 
 impl MessageQueue {
-    pub(crate) fn new() -> Self {
-        Self(HashMap::new())
-    }
-
-    /// Store a message in the MessageQueue.
-    ///
-    /// `identifier` corresponds to a unique session identifier.
-    pub(crate) fn store(
-        &mut self,
-        message_type: MessageType,
-        identifier: Identifier,
-        message: Message,
-    ) -> Result<()> {
-        let key = Self::get_key(message_type, identifier)?;
-        self.0.entry(key).or_default().push(message);
+    /// Store a message by the given [`MessageType`].
+    pub(crate) fn store(&mut self, message_type: MessageType, message: Message) -> Result<()> {
+        self.0.entry(message_type).or_default().push(message);
         Ok(())
     }
 
-    /// Retrieve (and remove) all messages of a given type from the
-    /// MessageQueue.
-    ///
-    /// `identifier` corresponds to a unique session identifier.
-    pub(crate) fn retrieve_all(
-        &mut self,
-        message_type: MessageType,
-        identifier: Identifier,
-    ) -> Result<Vec<Message>> {
-        self.do_retrieve(message_type, identifier, None)
+    /// Retrieve (and remove) all messages of a given [`MessageType`].
+    pub(crate) fn retrieve_all(&mut self, message_type: MessageType) -> Result<Vec<Message>> {
+        self.do_retrieve(message_type, None)
     }
 
-    /// Retrieve (and remove) all messages of a given type from a given sender
-    /// from the MessageQueue.
-    ///
-    ///`identifier` corresponds to a unique session identifier.
+    /// Retrieve (and remove) all messages of a given [`MessageType`] associated
+    /// with the given [`ParticipantIdentifier`].
     pub(crate) fn retrieve(
         &mut self,
         message_type: MessageType,
-        identifier: Identifier,
         sender: ParticipantIdentifier,
     ) -> Result<Vec<Message>> {
-        self.do_retrieve(message_type, identifier, Some(sender))
+        self.do_retrieve(message_type, Some(sender))
     }
-    ///`identifier` corresponds to a unique session identifier.
+
     fn do_retrieve(
         &mut self,
         message_type: MessageType,
-        identifier: Identifier,
         sender: Option<ParticipantIdentifier>,
     ) -> Result<Vec<Message>> {
-        let key = Self::get_key(message_type, identifier)?;
         // delete retrieved messages from storage so that they aren't accidentally
         // processed again.
-        let queue = self.0.remove(&key).unwrap_or_default();
+        let queue = self.0.remove(&message_type).unwrap_or_default();
```

Comment:
suggestion: I was going to ask what happens if something borks later in this function without completing processing on these messages, but it looks like there's actually no error case here. What do you think about having this (and the upstream `retrieve` methods) return a plain `Vec` instead of a `Result`? Is it worth keeping the result around in case we expand this implementation later to have actual error cases?

Reply:
Agreed. I think it's better to remove the `Result` and clean up the API now versus leave it around.